pinot-core/src/main/java/com/linkedin/pinot/core/data/extractors/PlainFieldExtractor.java

```diff
@@ -209,6 +210,15 @@ public GenericRow transform(GenericRow row, GenericRow destinationRow) {
             hasError = true;
             _errorCount.put(column, _errorCount.get(column) + 1);
           }
+        } else if (value instanceof ByteBuffer) { // TODO: Need better handle for ByteBuffers.
+          // ByteBuffer implementations are package private and cannot be put into TYPE_MAP.
+          ByteBuffer byteBuffer = (ByteBuffer) value;
+
+          // Assumes byte-buffer is ready to read. Also, avoid getting underlying array, as it may be over-sized.
```

Comment:
you can check if hasArray and use the bytebuffer.array to avoid creating byte[].